spec/lib/open_food_network/tag_rule_applicator_spec.rb

```diff
@@ -109,30 +109,20 @@ module OpenFoodNetwork
       end
     end
 
-    describe "filter!" do
+    describe "filter" do
       let(:applicator) { OpenFoodNetwork::TagRuleApplicator.new(enterprise, "FilterProducts", []) }
 
-      context "when the subject is nil" do
-        let(:subject) { double(:subject, reject!: false) }
-
-        it "returns immediately" do
-          applicator.filter!(subject)
-          expect(subject).to_not have_received(:reject!)
-        end
+      it "handles nil" do
+        applicator.filter(nil)
```

Comment:
I just wondering if we should also check if it returns an empty array ?

Reply:
Good point, I didn't think about that. I just adapted the old specs. It handles empty arrays as well though. So unless we discover a bug I wouldn't add that spec.